cmd/cleanup-jobs.go

```diff
@@ -0,0 +1,62 @@
+package cmd
+
+import (
+	"context"
+	"log"
+	"sync"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// CleanupJobs removes all failed jobs in a given namespace
+var CleanupJobs = &cobra.Command{
+	Use:   "cleanupjobs",
+	Short: "remove all failed jobs in a given namespace",
+	Run: func(cmd *cobra.Command, args []string) {
+
+		namespace, _ := cmd.Flags().GetString("namespace")
+
+		p := getPlatform(cmd)
+		clientSet, err := p.GetClientset()
+
+		check(err)
+
+		// gather the list of jobs from a namespace
+		jobs, err := clientSet.BatchV1().Jobs(namespace).List(context.TODO(), metav1.ListOptions{})
+
+		// WaitGroup to synchronize go routines execution
+		wg := sync.WaitGroup{}
+
+		// loop through the list of jobs found
+		for _, j := range jobs.Items {
+			wg.Add(1)
+
+			go func(j v1.Job, clientSet *kubernetes.Clientset) {
+
+				// if job status is not equal to "succeeded", delete the job
+				if j.Status.Failed == 0 {
+					log.Printf("INFO: Removing job %v from namespace %v", j.Name, j.Namespace)
+					err = clientSet.BatchV1().Jobs(namespace).Delete(context.TODO(), j.Name, metav1.DeleteOptions{})
+					check(err)
+				}
+				wg.Done()
+			}(j, clientSet)
+		}
+
+		wg.Wait()
+	},
+}
+
+func init() {
+	CleanupJobs.Flags().String("namespace", "", "Namespace to cleanup failed jobs.")
+}
+
+// function to print error message and abort program execution (os.Exit(1))
```

Comment:
Thank you for reviewing the code! :smile:

The `exit(1)` was not meant to be returned by the function. From the `log.Fatalf` doc:
> "Fatalf is equivalent to Printf() followed by a call to os.Exit(1)." 

Anyways, I'm removing this function to avoid confusions/mistakes and will follow the same pattern as from the project:
~~~
if err:=.... ; err != nil {
~~~

Reply:
I was not aware of that - I have always explicitly exited 1.  In that case, the check() function would not be appropriate for use on line 43.  Failing to delete one of a list of jobs should not be fatal